seisbench/models/base.py

```diff
@@ -1960,15 +2013,27 @@ def _trim_nan(x):
         Removes all starting and trailing nan values from a 1D array and returns the new array and the number of NaNs
         removed per side.
         """
-        mask = ~np.isnan(x)
-        valid = np.nonzero(mask)[0]
-        mask[valid[0] : valid[-1]] = True
-        _end = len(x)
-        x = x[mask]
+        nan_mask = ~np.isnan(x)
+        valid = np.nonzero(nan_mask)[0]
+        nan_mask[valid[0] : valid[-1]] = True
+        end = x.size
 
-        return x, valid[0], _end - (1 + valid[-1])
+        return x[nan_mask], valid[0], end - (1 + valid[-1])
 
-    def _recursive_torch_to_numpy(self, x):
+    @staticmethod
```

Comment:
This function seems unused. I think it's from the other PR? I'd suggest removing it here.